GREYErrorFormatter.m

```diff
@@ -0,0 +1,195 @@
+                                        indent:kGREYObjectFormatIndent
+                                     hideEmpty:YES
+                                      keyOrder:nil];
+}
+
+#pragma mark - Private Methods
+
+- (NSSet<NSString *> *)loggerKeys {
+  if ([_error.domain isEqualToString:kGREYInteractionErrorDomain] &&
+       _error.code == kGREYInteractionElementNotFoundErrorCode) {
+    return [[NSSet alloc] initWithArray:@[kErrorFormatterExceptionReasonKey,
+                                          kErrorFormatterRecoverySuggestionKey,
+                                          kErrorFormatterElementMatcherKey,
+                                          kErrorFormatterSearchActionInfoKey,
+                                          kErrorFormatterScreenshotPathsKey,
+                                          kErrorFormatterUnderlyingErrorKey,
+                                          kErrorFormatterUIHierarchyKey,
+                                          kErrorFormatterStackTraceKey]];
+  }
+  GREYFatalAssertWithMessage(false, @"Error Domain and Code Not Yet Supported");
+}
+
+- (NSString *)loggerDescriptionForKeys:(NSSet<NSString *> *)keys {
+  NSMutableArray<NSString *> *logger = [[NSMutableArray alloc] init];
+  
+  if ([keys containsObject:kErrorFormatterExceptionReasonKey]) {
+    NSString *exceptionReason = _error.localizedDescription;
+    [logger addObject:[NSString stringWithFormat:@"\n%@\n", exceptionReason]];
+  }
+  
+  if ([keys containsObject:kErrorFormatterRecoverySuggestionKey]) {
+    NSString *recoverySuggestion = _error.userInfo[kErrorDetailRecoverySuggestionKey];
+    if (recoverySuggestion) {
+      [logger addObject:[NSString stringWithFormat:@"%@\n", recoverySuggestion]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterElementMatcherKey]) {
+    NSString *elementMatcher = _error.userInfo[kErrorDetailElementMatcherKey];
+    if (elementMatcher) {
+      [logger addObject:[NSString stringWithFormat:@"%@:\n%@\n", kErrorDetailElementMatcherKey,
+                         elementMatcher]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterSearchActionInfoKey]) {
+    NSString *searchActionInfo = _error.userInfo[kErrorDetailSearchActionInfoKey];
+    if (searchActionInfo) {
+      [logger addObject:[NSString stringWithFormat:@"%@\n%@\n", kErrorDetailSearchActionInfoKey,
+                         searchActionInfo]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterScreenshotPathsKey]) {
+    for (NSString *key in _error.appScreenshots.allKeys) {
+      NSString *screenshotPath = _error.appScreenshots[key];
+      [logger addObject:[NSString stringWithFormat:@"%@: %@\n", key, screenshotPath]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterUnderlyingErrorKey]) {
+    NSString *nestedError = _error.nestedError.description;
+    if (nestedError) {
+      [logger addObject:[NSString stringWithFormat:@"%@: \n%@\n", kErrorFormatterUnderlyingErrorKey,
```

Comment:
redundant space in "%@: \n%@\n"?

Reply:
No, because this would say "Underlying Error: ..."

Reply:
You're right then - the string formatting is not very readable, so i made a new helper method for it